### DISCUSSION TITLE
Sujet / Thema 5 –– Cookbooks, Tutorials und Showcases für die Nutzung von Geodaten / Livres de recettes, tutoriels et vitrines pour l’utilisation des géodonnées

### DISCUSSION THREAD

**Discussion by rastrau** [GeoUnconference/discussions#4] [Ideen / Idées]
Resultate der Diskussion an der GeoUnconference 1 / Résultats de la discussion lors de la GeoUnconference 1

![Cookbooks, tutorials, and showcases](https://user-images.githubusercontent.com/4510815/124478578-9db29180-dda5-11eb-8d04-e5cb3b7ed0ff.jpg)

**Comment by davidoesch** [comment]
proposed tutorials ADRESSES

- [x] USECASE-Address: Convert a coordinates pair to an address (individuals) : https://github.com/GeoUnconference/discussions/issues/14  with excel https://www.geo.admin.ch/de/news/aktuell.detail.news.html/geo-internet/news2019/news20190307.html 
- [x] USECASE-Address: Convert multiple coordinates pairs to  addresses (individuals) : with excel https://www.geo.admin.ch/de/news/aktuell.detail.news.html/geo-internet/news2019/news20190307.html 
- [x] USECASE-Address: Convert an address to a coordinate pair  (individuals) : (a website) done with https://github.com/jonashofer/swisstopo-showcases 
- [x] USECASE-Address: Convert multiple addresses to coordinates pair  (individuals) : with excel , as in 
 https://opencagedata.com/tutorials/geocode-in-excel and today in https://www.geo.admin.ch/de/news/aktuell.detail.news.html/geo-internet/news2021/news20210331.html
- [x] USECASE- coordinates to Address  - ReverseGeocoding  / Find a address by Point (Developer) : Use API /URL template to get the address associated with latitude and longitude coordinates.  get inspired https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/find-a-location-by-point  and today in http://api.geo.admin.ch/services/sdiservices.html#examples-of-reverse-geocoding 
- [ ] USECASE- Address to coordinates - Geocoding:   Find a Coordinates by Query Address:  (Developer)  Use API / URL templates to get latitude and longitude coordinates that correspond to address information provided as a query string eg in https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/find-a-location-by-query and today in http://api.geo.admin.ch/services/sdiservices.html#id15  and python examples as documented in Chapter 6 ofhttps://www.cadastre.ch/content/cadastre-internet/de/services/service/registry/building/_jcr_content/contentPar/tabs_copy/items/dokumente/tabPar/downloadlist/downloadItems/317_1614004354008.download/Spezifikation-DE.pdf 

next step:

@pauloamado will

- [ ]  create based on https://github.com/GeoUnconference/discussions/issues/13 for each a description
- [ ] we will add them to the landing page of www.geo.admin.ch

**Reply by davidoesch** [reply]
@pauloamado  lie lien intern for all the address use cases is https://jira.swisstopo.ch/browse/KOGIS_PRJ-1280

**Reply by rastrau** [reply]
Cool thanks! @davidoesch can you please explain a bit more what the idea behind the collection of use cases and the use case template is:
- Do you invite others to fill in some of the use cases or will you fill them in yourself?
- Can others add their own desired use cases?
- What happens with the list of use cases once ~complete?

**Reply by davidoesch** [reply]
_Do you invite others to fill in some of the use cases or will you fill them in yourself?
Can others add their own desired use cases?_

 -> Everyone can /should bring in use  cases. Letrs try to group them, allign them for the different target groups (Indivualds, expert, developers) and different uses cases (eg addresses, dowload,  We will try to aggregate  them together. 
-> creating a ticket via "Create issue from discussion" on the right side and using the structure as in _TEMPLATE_ USECASE_SHOWCASE_TUTORIAL #13  should allow the group here to have a clear structure when formulating the use case /showcase

"What happens with the list of use cases once ~complete?"
-> from the geo.admin.ch side: we will test the group for "addresses "https://github.com/GeoUnconference/discussions/discussions/4#discussioncomment-978849  in www.geo.admin.ch as a test and invite the users for feedback ---and from there on we hope to get feedback and next steps in geounconference2

**Comment by rastrau** [comment]
Statt von der Use-Case-Seite von der Angebotsseite her gedacht würde mich interessieren, welche Tutorials, "Cookbooks" und Showcases es bereits gibt mit Geodaten von Schweizer Behörden? Ich denke dabei an:
- erläuternde Artikel
- Code und Dokumentation (z.B. in [Jupyter](https://jupyter.org) oder in [Observable](https://observablehq.com/))
- Wikis
- etc.

Aufgefallen sind mir u.a. die folgenden Inhalte:

**Open Data Zürich**
Open Data Zürich (Stadt Zürich, mit @dongoginho und @metaodi) pflegen die [Open Data-Werkstatt](https://www.stadt-zuerich.ch/portal/de/index/ogd/werkstatt.html). Darin sind verzeichnet bzw. verlinkt:
  - eine Erläuterung von Datenformaten (viel davon räumliche): u.a. GeoJSON, XML, REST-Services, WMS, WMTS, WFS, Shapefile, GeoPackage, DXF
  - [Tutorials zur Arbeit mit Daten](https://opendatazurich.github.io/) auf GitHub, u.a. mit Inhalten zum Geoportal:
    - Abfragen von WFS
    - Einbinden von WMS/WMTS in Leaflet
    - Nutzen von WFS aus Jupyter mit Python
  - [Crashkurs zu Datenvisualisierung](https://opendatazurich.github.io/crashkurs-dataviz/) (nicht räumlich)
  - [Crashkurs zu Power BI](https://opendatazurich.github.io/crashkurs-dataviz-powerbi/) (nicht räumlich)
Zudem gibt es [Showcases von Anwendungen von offenen Daten der Stadt Zürich](https://www.stadt-zuerich.ch/portal/de/index/ogd/anwendungen.html). Viele davon verwenden (auch) Geodaten.

**Open Data Basel-Stadt**
Das Open Data-Team von Basel-Stadt um @jb3-2 hat Showcases von [Anwendungen](https://www.opendata.bs.ch/anwendungen/datenvisualisierungen.html), [Datenjournalismus](https://www.opendata.bs.ch/anwendungen/datenjournalismus.html) und [Applikationen und Apps](https://www.opendata.bs.ch/anwendungen/applikationen-apps.html), viele davon mit räumlichen Daten.

Ich glaube, daneben gibt es allenfalls weitere Ressourcen von der FHNW (@martinchristen), Kanton Zürich (@openzh) oder OST? Kennt jemand von Euch noch weitere Tutorials, Cookbooks und Showcases? → gerne ergänzen 👍

**Comment by nrohrbach** [comment]
**Bundesamt für Energie**
Showcases: https://www.bfe.admin.ch/bfe/de/home/versorgung/statistik-und-geodaten/geoinformation/storymaps.html
oder auf opendata.swiss: https://opendata.swiss/de/showcase

Cookbook auf github für ichtankestrom.ch
https://github.com/SFOE/DIEMO-Documentation

**Comment by davidoesch** [comment]
**Für Addressen / Suchdienste**

POST
- https://developer.post.ch/en/address-web-services-soap 

BFS
- https://www.housing-stat.ch/de/query.html 

Google
- https://developers.google.com/maps/documentation/geocoding/overview 

Bing
- https://docs.microsoft.com/en-us/bingmaps/rest-services/locations/ ( IMHO very good)

Opencage
- https://opencagedata.com/  especiall the libraries. IMHO best solution for DEV

Stadiamaps
- locate service https://valhalla.readthedocs.io/en/latest/api/locate/api-reference

**Dienste Beschrieb**

Kt BS 

- http://www.geo.bs.ch/geodaten/geodienste/wmsbs.html 
- http://www.geo.bs.ch/geodaten/geodienste/wmtsbs.html

Sentinel
Service description: 

- https://www.sentinel-hub.com/develop/api/ogc/standard-parameters/wms/
examples

- https://www.sentinel-hub.com/develop/api/ogc/ogc-examples/ 

Google elevation API

- https://developers.google.com/maps/documentation/elevation/intro

GPS Visualizer  elevation

- https://www.gpsvisualizer.com/elevation 

BING Elevation

- https://docs.microsoft.com/en-us/bingmaps/rest-services/elevations/get-elevations 

IGN:
- DESKTOP  https://geoservices.ign.fr/documentation/utilisation_SIG/index.html 
- https://geoservices.ign.fr/documentation/utilisation_web/index.html für Openlayers  zb in https://geoservices.ign.fr/documentation/utilisation_web/wmts-web.html oder https://geoservices.ign.fr/documentation/utilisation_web/bib-acces.html 


Stadiamaps:

- Service Documentation https://docs.stadiamaps.com/ 
- Routing: https://docs.stadiamaps.com/routing/ 
- Elevation: https://valhalla.readthedocs.io/en/latest/api/elevation/api-reference/#outputs-of-the-elevation-service

**Reply by procrastinatio** [reply]
**Elevation** (SwissAlti3D https://www.swisstopo.admin.ch/en/geodata/height/alti3d.html)

**Current services**

https://api3.geo.admin.ch/height          (http://api3.geo.admin.ch/services/sdiservices.html#height)
https://api3.geo.admin.ch/profile	      (http://api3.geo.admin.ch/services/sdiservices.html#profile)

**Limitations/issues**

1/ Two endpoints.
Points and lines are not treated the same way, both as the service parameters and the results.

2/ Hard to attributes a height to a line or set of points
For line, the input line may (or not) be simplified (Douglas-Peuckert style) and a JSON containing a distance and a height along the original line returned (No OGD back then, simple solution to "protect" the costly DEM data)

3/ Distance in WebMercator or WGS1984?
What should we use?

**Possible uses cases**

_Elevation_
Give or check the height of a given point, set of points, line or a file (GPX, KML, GeoJSON)

_Profile_
Given a line as parameters, determine the inflection points on the DEM, densify the line and return, either 

_Hiking time_
On a footpath, hiking time should be extracted from a database, not calculated. Hiking time outside the footpath is only indicative (many unknown possible obstacles)


Currently existing services


Bing: GetElevation (https://docs.microsoft.com/en-us/bingmaps/rest-services/elevations/get-elevations)

Google: elevation API (https://developers.google.com/maps/documentation/elevation/overview)

Valhalla: Elevation API (https://valhalla.readthedocs.io/en/latest/api/elevation/api-reference/)

**Reply by davidoesch** [reply]
thx @procrastinatio 
I added them directly as task so we can work on themnot only as tutorial but also as service requirements

- https://github.com/GeoUnconference/discussions/issues/19
- https://github.com/GeoUnconference/discussions/issues/20
- https://github.com/GeoUnconference/discussions/issues/21
- https://github.com/GeoUnconference/discussions/issues/22

**Comment by dkottmann** [comment]
@rastrau 

> Statt von der Use-Case-Seite von der Angebotsseite her gedacht würde mich interessieren, welche Tutorials, "Cookbooks" und Showcases es bereits gibt mit Geodaten von Schweizer Behörden? Ich denke dabei an:
>
> -  erläuternde Artikel
> -  Code und Dokumentation (z.B. in Jupyter oder in Observable)
> -  Wikis
> -  etc.

Idee: Kurzes Video mit skizzenhafter Animation der Anwendungsmöglichkeiten (Anwendungsfälle, Vor- und Nachteile) verschiedener standardisierter Formate und Dienste

**Reply by rastrau** [reply]
Stimmt, ist mittlerweile ja ein verbreitetes "Anleitungsformat" / True, this is a frequent form of instruction by now, e.g., DIY or cooking tutorials.

**Reply by davidoesch** [reply]
@pauloamado was wir gerade besprochen haben: video tutorials.   4 stück, rein für den Kartenviewer  https://help.geo.admin.ch/?ids=01&lang=de  , 4 Sprachen Aufwände sind relativ hoch (Ersterstellung und Betrieb) aber es lont sich

**Reply by rastrau** [reply]
Ich persönlich nutze Anleitungsvideos nicht sehr gerne, aber mir ist bewusst, dass es vielen anders geht. Ich würde folgende formale Anforderungen erwägen, manche sicher nice-to-have (*diese könnten wir an der GeoUnconference #2 verifizieren / verfeinern, mit weiteren Anforderungen ergänzen!*):

- kurz (ev. für "Mikrofunktionen" auch als GIF), modular: kurze Videos, die kleine Aspekte erläutern
- oder wenn lang / länger: schneller abspielbar (z.B. 1.5-fache Geschwindigkeit), komfortabel skipbar, mit Timestamps versehen
- mobile-fähig, d.h. ev. quadratisches Format für Nutzung auf dem aufrecht gehaltenen Smartphone / Tablet
- untertitelt oder untertitelbar → bessere Accessibility und auch lautlos konsumierbar (ähnlich wie Videos auf Newsplattformen oder Twitter-Videos von SRF)

**Reply by hinjud** [reply]
Mir geht es gleich wie @rastrau. Die Videos sollten so kurz wie möglich sein und wirklich nur einen Einblick geben, mehr nicht. Vertieftere Informationen könnten über ein schriftliches Dokument abgegeben werden, welches, wo nötig, mit Printscreens oder allenfalls Links auf weitere Videos ergänzt wird. Wichtig ist auch, dass trotz Performanz-Anforderungen nicht zu stark an der Qualität der Video-Auflösung gespart wird, da dies aus meiner Sicht das "Erlebnis" wie auch den Eindruck der Professionalität schmälert. Eventuell könnte die Offline-Verfügbarkeit noch geprüft werden.

**Comment by schenardi** [comment]
Von Cookbooks würde ich erwarten, dass sei ein reales Problem und in Summe den ganzen Prozess beschreiben.
Es ist wenig zielführend, wenn z.B. aufgezeigt wird, wie in Jupyter eine Karte mit Punkten angezeigt wird aber offen bleibt, wie diese Daten letztendlich als Anwendung oder Dienst bereitgestellt werden können.

Ebenso sollte darauf geachtet werden, dass die meist verbreitetsten Technologien und Anwendungen abgedeckt werden (die üblichen Desktop-Clients, Einbindung als Service in XaaS-Lösungen wie ArcGIS Online, Mapbox, etc und auch Aufbereitung für Self-Hosting on Premise oder in der Cloud.)

**Comment by jonashofer** [comment]
Als Semesterprojekt an der Berner Fachhochschule haben wir uns in Absprache mit @davidoesch dem Usecase **"Convert an address to a coordinate pair (individuals) : (a website)"** angenommen und freuen uns nun, das Resultat zu veröffentlichen.

Repository: https://github.com/jonashofer/swisstopo-showcases
URL: https://swisstopo-showcases.netlify.app/

Wir freuen uns über Feedback.

**Comment by davidoesch** [comment]
USECASE-Address: Convert an address to a coordinate pair (individuals) : (a website) done  https://github.com/jonashofer/swisstopo-showcases

**Comment by davidoesch** [comment]
FYI: GEOLOCATION API IS A W3C RECOMMENDATION https://www.w3.org/blog/news/archives/9660
